From a141a26e9ee8d347a977d7a1d3711f5ccf97247f Mon Sep 17 00:00:00 2001
From: Uday Bondhugula <uday@csa.iisc.ernet.in>
Date: Thu, 23 Mar 2017 16:13:28 +0530
Subject: [PATCH 2/2] isl dim-wise single-valued and isl_pw_aff_map functions

Signed-off-by: Uday Bondhugula <uday@csa.iisc.ernet.in>
---
 isl_map.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/isl_map.c b/isl_map.c
index 8ac919b..a1e0302 100644
--- a/isl_map.c
+++ b/isl_map.c
@@ -9829,10 +9829,9 @@ __isl_give isl_map *isl_map_range_factor_domain(__isl_take isl_map *map)
 	return map;
 }
 
-
-/* Given a basic_map A -> [B -> C], extract the basic_map A -> B.
- */
-__isl_give isl_basic_map *isl_basic_map_range_factor_domain(__isl_take isl_basic_map *bmap)
+/* Given a basic map A -> [B -> C], extract the basic map A -> B. */
+__isl_give isl_basic_map *isl_basic_map_range_factor_domain(__isl_take 
+        isl_basic_map *bmap)
 {
 	isl_space *space;
 	int total, keep;
@@ -9853,7 +9852,6 @@ __isl_give isl_basic_map *isl_basic_map_range_factor_domain(__isl_take isl_basic
 	return bmap;
 }
 
-
 /* Given a map A -> [B -> C], extract the map A -> C.
  */
 __isl_give isl_map *isl_map_range_factor_range(__isl_take isl_map *map)
-- 
2.7.4

