From cc57331175b269dfb047784bf45dc33de9baf85f Mon Sep 17 00:00:00 2001
From: Chandan Reddy <chandan1986g@gmail.com>
Date: Wed, 7 Dec 2016 18:35:30 +0100
Subject: [PATCH] added extra braces to prevent the redefination error with omp
 parallel loops

---
 source/pprint.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source/pprint.c b/source/pprint.c
index b9913af..21d1f07 100644
--- a/source/pprint.c
+++ b/source/pprint.c
@@ -474,6 +474,7 @@ void pprint_for(struct cloogoptions *options, FILE *dst, int indent,
                     (f->time_var_name) ? f->time_var_name : "");
         }
         if ((f->parallel & CLAST_PARALLEL_OMP) && !(f->parallel & CLAST_PARALLEL_MPI)) {
+	    fprintf(dst, "{/* extra braces to avoid redefination of lbp*/\n");
             if (f->LB) {
                 fprintf(dst, "int lbp%s=", f->suffix);
                 pprint_expr(options, dst, f->LB);
@@ -639,6 +640,9 @@ void pprint_for(struct cloogoptions *options, FILE *dst, int indent,
     else
 	fprintf(dst,"}\n") ; 
 
+        if ((f->parallel & CLAST_PARALLEL_OMP) && !(f->parallel & CLAST_PARALLEL_MPI)) {
+		fprintf(dst,"}/*end of omp parallel loop */\n") ; 
+}
     if (options->language == CLOOG_LANGUAGE_C) {
         if (f->time_var_name) {
             fprintf(dst, "IF_TIME(%s += cloog_util_rtclock() - %s_start);\n",
-- 
1.9.1

