diff --git a/include/pet.h b/include/pet.h
index ed24e01..250552f 100644
--- a/include/pet.h
+++ b/include/pet.h
@@ -169,6 +169,7 @@ struct pet_stmt {
 	isl_set *domain;
 	isl_map *schedule;
 	struct pet_expr *body;
+        char * stmt_text;
 
 	unsigned n_arg;
 	struct pet_expr **args;
@@ -345,6 +346,9 @@ __isl_give isl_union_map *pet_scop_collect_must_kills(struct pet_scop *scop);
 __isl_give isl_union_map *pet_scop_collect_tagged_must_kills(
 	struct pet_scop *scop);
 __isl_give isl_union_map *pet_scop_collect_schedule(struct pet_scop *scop);
+__isl_give isl_union_map *pet_stmt_collect_accesses(struct pet_stmt *stmt,
+	int read, int write, int kill, int must, int tag,
+	__isl_take isl_space *dim);
 
 #if defined(__cplusplus)
 }
diff --git a/scop.c b/scop.c
index 81c2f14..fd59414 100644
--- a/scop.c
+++ b/scop.c
@@ -3638,7 +3638,7 @@ static __isl_give isl_union_map *scop_collect_accesses(struct pet_scop *scop,
 		accesses = isl_union_map_union(accesses, accesses_i);
 	}
 
-	arrays = isl_union_set_empty(isl_union_map_get_space(accesses));
+	/*arrays = isl_union_set_empty(isl_union_map_get_space(accesses));
 	for (i = 0; i < scop->n_array; ++i) {
 		isl_set *extent = isl_set_copy(scop->arrays[i]->extent);
 		arrays = isl_union_set_add_set(arrays, extent);
@@ -3646,7 +3646,7 @@ static __isl_give isl_union_map *scop_collect_accesses(struct pet_scop *scop,
 	accesses = isl_union_map_intersect_range(accesses, arrays);
 
 	to_inner = compute_to_inner(scop);
-	accesses = isl_union_map_apply_range(accesses, to_inner);
+	accesses = isl_union_map_apply_range(accesses, to_inner);*/
 
 	return accesses;
 }
@@ -3679,6 +3679,14 @@ __isl_give isl_union_map *pet_scop_collect_must_kills(struct pet_scop *scop)
 	return scop_collect_accesses(scop, 0, 0, 1, 1, 0);
 }
 
+/* Function for statement accesses required in pluto's program.c*/
+__isl_give isl_union_map *pet_stmt_collect_accesses(struct pet_stmt *stmt,
+	int read, int write, int kill, int must, int tag,
+	__isl_take isl_space *dim)
+{
+        return stmt_collect_accesses(stmt, read, write, kill, must, tag, dim);
+}
+
 /* Collect all tagged potential read access relations.
  */
 __isl_give isl_union_map *pet_scop_collect_tagged_may_reads(
